### PR TITLE
[libc++][doc] Add warning about limitation of --fresh in boostrapping build

### PR DESCRIPTION
**Pull request by hawkinsw**
Add a warning to the `Building Libcxx` documentation about the limitations of the utility of `--fresh` at the top level.

### CODE REVIEW

**Comment by hawkinsw** [comment]
I hope that my lost time saves someone else! 

cc @ldionne @ericwf (based on `git-blame` -- sorry if I inaccurately tagged!).

File: libcxx/docs/BuildingLibcxx.rst [hunk 1]
```diff
@@ -78,6 +78,9 @@ CMake invocation at ``<monorepo>/llvm``:
   This type of build is also commonly called a "Runtimes build", but we would like to move
   away from that terminology, which is too confusing.
 
+.. warning::
+  Adding the `--fresh` flag to the top-level cmake invocation in a bootstrapping build *will not* freshen the cmake cache of any of the enabled runtimes.
```
Comment (ldionne) [inline libcxx/docs/BuildingLibcxx.rst:82]:
```suggestion
  Adding the `--fresh` flag to the top-level cmake invocation in a bootstrapping build *will not*
  freshen the cmake cache of any of the enabled runtimes.
```

Comment (ldionne) [reply libcxx/docs/BuildingLibcxx.rst:82]: Let's just try to break the line similarly to the other lines in this file.

**Comment by ldionne** [comment]
LGTM with a small nitpick.

**Review by ldionne — APPROVED** [review]

File: libcxx/docs/BuildingLibcxx.rst (hunk 1 above)
Comment (hawkinsw) [reply libcxx/docs/BuildingLibcxx.rst:82]: Absolutely!! Thanks! Fixup coming!

**Comment by ldionne** [comment]
@hawkinsw I have the option to squash and merge on your behalf here but it says the commit will be authored by 8715530+hawkinsw@users.noreply.github.com. I assume that's not what you want right?

**Comment by hawkinsw** [comment]
@ldionne I believe that I have fixed it now! You should see a drop-down that asks which email to associate with the commit. I would prefer `whh8b@obs.cr`. Thank you for your patience!!

PS: In case you are wondering, the cause of this problem was that I had my verified email addresses set to _hidden_ in my GitHub profile. As a result, GitHub was attempting to keep people from seeing my full address in public in the commit log. That is a nice feature, but the UI seems a little odd. I hope that information helps if you run into similar issues in the transition to GitHub workflows.

**Comment by ldionne** [comment]
Thanks for the explanation, this is good to know!

**Pull request merged**